[Kolors] Add IP Adapter (#8901)
huggingface/diffusers: docs/source/en/api/pipelines/kolors.md

image.save("kolors_sample.png")
```

> ## IP Adapter

Reviewer: I think it might be clearer to nest this under Usage Example (ie, IP-Adapter is a usage example and not something separate).

```suggestion
### IP Adapter
```

> Kolors needs a different IP Adapter to be able to work, also it uses Openai-CLIP-336 as an image encoder.

Reviewer: Maybe link to the model here https://huggingface.co/openai/clip-vit-large-patch14-336

```suggestion
Kolors needs a different IP Adapter to work, and it uses Openai-CLIP-336 as an image encoder.
```

<Tip>

> Using an IP Adapter with Kolors needs more than 24GB of VRAM, it is recommended to use `enable_model_cpu_offload()` in consumer GPUs to be able to use it.

Reviewer: ```suggestion
Using an IP Adapter with Kolors requires more than 24GB of VRAM. To use it, we recommend using [`~DiffusionPipeline.enable_model_cpu_offload`] on consumer GPUs.
```

</Tip>

<Tip>

> While the PR is merged, we need to give the revision and load the image encoder separatedly to use the safetensors format. You can still use the main branch of the original repository if you're comfortable loading `pickle` checkpoints.

Reviewer: ```suggestion
While Kolors is integrated in Diffusers, you need to load the image encoder from a revision to use the safetensor files. You can still use the main branch of the original repository if you're comfortable loading pickle checkpoints.
```
